x/sequencer/proposal/tx_validator_data_item_test.go

```diff
@@ -0,0 +1,147 @@
+package proposal
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/warp-contracts/sequencer/x/sequencer/test"
+	"github.com/warp-contracts/sequencer/x/sequencer/types"
+)
+
+func validatorLoggerAndMsg(t *testing.T) (*TxValidator, *LoggerMock, types.MsgDataItem) {
+	arweaveBlockInfo := test.ArweaveBlock().BlockInfo
+	lastBlock := &types.LastArweaveBlock{
+		ArweaveBlock:         arweaveBlockInfo,
+		SequencerBlockHeight: 123,
+	}
+	validator, logger := validatorAndLogger(t, lastBlock, arweaveBlockInfo, nil)
+	msg := test.ArweaveL2Interaction(t)
+	return validator, logger, msg
+}
+
+func TestCheckSortKey(t *testing.T) {
+	validator, logger, msg := validatorLoggerAndMsg(t)
+	msg.SortKey = "000001431216,0000000000123,00000000"
+
+	result := validator.checkSortKey(&msg)
+
+	require.True(t, result)
+	require.Equal(t, "", logger.Msg)
+}
+
+func TestCheckSortKeyNoSortKey(t *testing.T) {
+	validator, logger, msg := validatorLoggerAndMsg(t)
+
+	result := validator.checkSortKey(&msg)
+
+	require.False(t, result)
+	require.Equal(t, "Rejected proposal: invalid sort key", logger.Msg)
+}
+
+func TestCheckSortKeyInvalidArweaveBlock(t *testing.T) {
+	validator, logger, msg := validatorLoggerAndMsg(t)
+	msg.SortKey = "000001431217,0000000000123,00000000"
+
+	result := validator.checkSortKey(&msg)
+
+	require.False(t, result)
+	require.Equal(t, "Rejected proposal: invalid sort key", logger.Msg)
+}
+
+func TestCheckSortKeyInvalidSequencerBlock(t *testing.T) {
+	validator, logger, msg := validatorLoggerAndMsg(t)
+	msg.SortKey = "000001431216,0000000000124,00000000"
+
+	result := validator.checkSortKey(&msg)
+
+	require.False(t, result)
+	require.Equal(t, "Rejected proposal: invalid sort key", logger.Msg)
+}
+
+func TestCheckSortKeyInvalidIndex(t *testing.T) {
+	validator, logger, msg := validatorLoggerAndMsg(t)
+	msg.SortKey = "000001431216,0000000000123,00000001"
+
+	result := validator.checkSortKey(&msg)
```

Comment:
Maybe it should return an error. This way we won't have to check error messages in tests.